### ISSUE TITLE
Improve access to /mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer using user with READ-ONLY access-rights

### ISSUE THREAD

**Issue by joebride** [F5Networks/f5-telemetry-streaming#76]
### Is your feature request related to a problem? Please describe.
Grafana/Prometheus are tools used by my customer, which should get access to my BIG-IP statistics.
According to F5-documentation (and my test-results) this must be done per REST-API https://<ip-of-BIG-IP>/mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer **using Account with ADMIN-rights.**

This means a Prometheus user-account must be created on my BIG-IP with ADMIN-rights!
So **customer gets full control over my BIG-IP**, which is really bad and not allowed by my security-policy!

### Describe the solution you'd like
Access to https://<ip-of-BIG-IP>/mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer should be modified in the following way:
1. possible with **READ-ONLY** account
2. allow this specific path only with some **path filter-mechanism**

### Describe alternatives you've considered
n/a

### Additional context
Declaration is as follows - taken from "https://github.com/mdditt2000/prometheus"
```json
{
    "class": "Telemetry",
    "My_Poller": {
        "class": "Telemetry_System_Poller",
        "interval": 0
    },
    "My_System": {
        "class": "Telemetry_System",
        "enable": "true",
        "systemPoller": "My_Poller"
    },
    "My_Pull_Consumer": {
        "class": "Telemetry_Pull_Consumer",
        "type": "Prometheus",
        "systemPoller": "My_Poller"
    }
}
```

**Comment by dstokesf5** [comment]
I was able to create a user that only had read access to the `mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer ` endpoint by following this guide: https://devcentral.f5.com/s/articles/icontrol-rest-fine-grained-role-based-access-control-30773

Below is the resource group I created for my prometheus user:
```json
{
    "name": "prometheusResourceGrup",
    "resources": [
        {
            "restMethod": "GET",
            "resourceMask": "/mgmt/shared/telemetry/pullconsumer/My_Pull_Consumer"
        }
    ]
}
```

**Comment by joebride** [comment]
yes, it works for me too, using GUEST-role.
Is there a plan to add this RBAC feature into **tmsh**?

**Comment by dstokesf5** [comment]
I do not have any insight on the road map for this RBAC functionality.

**Comment by dstokesf5** [comment]
Closing for inactivity. If you still have an issue, please re-open this issue or create a new one.

**Comment by jgrig90** [comment]
Is it possible to setup an account with guest access for telemetry when you have LDAP for authentication setup? Would you need to set the account up as local or in AD to achieve this?

**Comment by dstokesf5** [comment]
Unfortunately my knowledge on this part of BIG-IP is mostly limited to the article I posted previously: https://devcentral.f5.com/s/articles/icontrol-rest-fine-grained-role-based-access-control-30773. You could try to use an LDAP user for step 4.

**Comment by jgrig90** [comment]
Thanks for the reply. I was able to figure out that having both an AD service account and local account created with the same Username and AD password allows for the steps in the devcentral article above to work. This way you don't have to add any Remote Role Groups if you're not adding the AD account to anything else except for the Domain Users group. If anyone finds this to be confusing as I did when setting this up feel free to reach out to me.
